core/src/read.jl

```diff
@@ -635,111 +635,143 @@ function PidControl(db::DB, config::Config, chunk_sizes::Vector{Int})::PidContro
     )
 end
 
-function UserDemand(db::DB, config::Config)::UserDemand
-    static = load_structvector(db, config, UserDemandStaticV1)
-    time = load_structvector(db, config, UserDemandTimeV1)
-
-    static_node_ids, time_node_ids, node_ids, valid =
-        static_and_time_node_ids(db, static, time, "UserDemand")
+function user_demand_static!(
+    active::BitVector,
+    demand::Vector{Float64},
+    demand_itp::Vector{Vector{ScalarInterpolation}},
+    return_factor::Vector{Float64},
+    min_level::Vector{Float64},
+    static::StructVector{UserDemandStaticV1},
+    node_ids::Vector{NodeID},
+    priorities::Vector{Int32},
+)::Nothing
+    user_demand_idx = 1
```

Comment:
Instead of user_demand_idx = 1 and the while nodes, I recommend this:

```julia
indices = searchsorted(node_ids, unique(static.node_id)
grouped = IterTools.groupby(row -> row.node_id, static)
for (user_demand_idx, group) in zip(indices, grouped)
     ...
end

Reply:
Is there a particular reason why you want to pre-search the indices, and not search them within the loop?